### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by Bodigrim**
This is in preparation for `tasty-1.4.3` (#355) release.

### CODE REVIEW

File: core/CHANGELOG.md [hunk 1]
```diff
@@ -6,23 +6,32 @@ Unreleased
...
+* Dependency loop error now lists all test cases that formed a cycle
+
+Version 1.4.3
+---------------
+
+_2022-12-18_
+
+* Drop `unbounded-delays` dependency on 64-bit machines.
+* Drop `clock` dependency, decomission `clock` flag.
+* Drop `wcwidth` dependency.
```
Comment (andreasabel) [inline core/CHANGELOG.md:18]: Please add a "Tested with" GHC version range here.

File: core/CHANGELOG.md [hunk 2]
```diff
@@ -6,23 +6,32 @@ Unreleased
...
-- Dependency loop error now lists all test cases that formed a cycle
+* Dependency loop error now lists all test cases that formed a cycle
+
+Version 1.4.3
+---------------
+
+_2022-12-18_
+
+* Drop `unbounded-delays` dependency on 64-bit machines.
+* Drop `clock` dependency, decomission `clock` flag.
```
Comment (andreasabel) [inline core/CHANGELOG.md:17]: If this has an issue, please link to it.

**Review by andreasabel — CHANGES_REQUESTED** [review]
Thanks, LG, with some suggestions.

**Comment by Bodigrim** [comment]
@andreasabel updated.

**Comment by Bodigrim** [comment]
@andreasabel I believe I fixed your remarks, so I'll go ahead with a merge to match the proposed date for 1.4.3 release. Happy to make further amends later if you get more suggestions.

**Pull request merged**